src/components/Form/index.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
+
+import type {
+    Comment,
+    CommentReply,
+} from '../../state/comments';
+
+
+export interface CommentFormSetProps {
+    comments: Comment[];
+}
+
+export function CommentFormSetComponent(props: CommentFormSetProps) {
+    const [initialNumber] = React.useState(props.comments.filter(comment => {return comment.remoteId != null}).length);
+    const prefix = "comments";
+
+    const commentForms = props.comments.map((comment, formNumber) => (
+        <CommentFormComponent
+            key={comment.localId}
+            comment={comment}
+            formNumber={formNumber}
```

Comment:
**Nitpick:** Wouldn't bank on using array index if you want this to be consistent between re-renders and adding new data. I.e. you need to be careful with ordering and just remember this is how it works I guess. If this is all dynamic and can be, that's great. Although I am not sure if there's better way in this particular context.

Reply:
Hmm. I was aware there were problems with using index as a key, but I'm avoiding that. Not sure exactly what you mean - that a component's index prop may change between renders? If so, I think that's fine, ordering is pretty irrelevant here, I just need the formset's individual forms to have sequential numbers, but whether that sequence stays the same doesn't matter.